pock/dock-widget
# fixed teams app issues

---
This MR fixes multiple Teams issues that are caused by the Team Helper.
---
_This is a rework of pock/pock#410._

Teams has a dedicated App to show the notifications (called `Microsoft Teams Helper`).
It has the same bundleId, but a different name and bundleUrl.
This leads to some bugs (pock/pock#20, pock/pock#108, pock/pock#158, pock/pock#228, https://github.com/pigigaldi/Pock/issues/108#issuecomment-508012486)

I fixed these by also comparing apps by their bundleURL, but not only by their bundleId.

This seems to fix 4 Bugs:
- Wrong Teams Icon (see pock/pock#20, pock/pock#108, pock/pock#158)
- Jumping Teams Icon (pock/pock#108)\
_The reason was the start of the teams helper._
- Crashing Teams when clicked (pock/pock#228)\
_The reason was the running of the teams helper._
- Missing badge (when the Helper is running) (https://github.com/pigigaldi/Pock/issues/108#issuecomment-508012486)

** Screenshot**
<img width="1004" alt="Touch Bar Bild 2020-05-21 um 11 59 08" src="https://user-images.githubusercontent.com/11719855/82547730-7d505500-9b5a-11ea-8c96-52196b16b9e0.png">


**Fixes**
pock/pock#20, pock/pock#108, pock/pock#158, pock/pock#228

**Testing**
*Test Configuration*:
* Hardware:           MacBook Pro, 16 (2019)
* macOS Version: 10.15.4
* Xcode version:   12.4 (12D4e)

1. Start Build
2. Open Teams
3. Get somebody to send you a message
4. See if Badge is still there and Icon is correct
5. Click on Teams Icon (while Teams Helper is running)
6. be satisfied 🎉

_You can check if the teams helper is running, via right-clicking on the dock icon of teams._
